test/Datadog.Trace.TestHelpers/MockTracerAgent.cs

```diff
@@ -62,7 +63,7 @@ public MockTracerAgent(int port = 8126, int retries = 5)
 
         public IImmutableList<Span> Spans { get; private set; } = ImmutableList<Span>.Empty;
 
-        public IImmutableList<HttpListenerRequest> RawRequests { get; private set; } = ImmutableList<HttpListenerRequest>.Empty;
+        public IImmutableList<NameValueCollection> RequestHeaders { get; private set; } = ImmutableList<NameValueCollection>.Empty;
```

Comment:
Was this causing problems as HttpListenerRequest?

Reply:
Yes, it was. On .NET Framework, when the Response was closed (and sent), that object AND the corresponding Request object were disposed and inaccessible.